coder.ts

```diff
@@ -0,0 +1,219 @@
+            debug(`Found file with code component: ${relativePath}`);
+            return res;
+          }
+          debug(
+            `Skipping file (no code component with file meta): ${relativePath}`,
+          );
+        }
+
+        return [];
+      }),
+    );
+    return files.flat();
+  };
+
+  return walkDir(directory);
+};
+
+const processContent = async (
+  content: string,
+  filePath: string,
+): Promise<string> => {
+  try {
+    const file = await unified()
+      .use(remarkParse)
+      .use(remarkFrontmatter)
+      // @ts-expect-error
+      .use(codeImport, {
+        preserveTrailingNewline: false,
+        removeRedundantIndentations: true,
+        rootDir: process.cwd(),
+      })
+      .use(remarkStringify, {
+        bullet: "-",
+        emphasis: "*",
+        fences: true,
+        listItemIndent: "one",
+        rule: "-",
+        ruleSpaces: false,
+        strong: "*",
+        tightDefinitions: true,
+      })
+      .process(content);
+    return String(file);
+  } catch (error) {
+    if ((error as NodeJS.ErrnoException).code === "ENOENT") {
+      throw new Error(
+        `File not found: ${(error as NodeJS.ErrnoException).path}`,
+      );
+    }
+    throw error;
+  }
+};
+
+const processFile = async (filePath: string): Promise<void> => {
+  try {
+    if (!(await hasCodeComponentWithFileMeta(filePath))) {
+      debug(`Skipping ${filePath}: No code component with file meta found.`);
+      return;
+    }
+
+    const originalContent = await fs.readFile(filePath, "utf8");
+    const processedContent = await processContent(originalContent, filePath);
+    if (originalContent !== processedContent) {
+      await fs.writeFile(filePath, processedContent);
+      console.log(`Updated: ${filePath}`);
+    } else {
+      debug(`No changes needed for: ${filePath}`);
+    }
+  } catch (error) {
+    console.error(`Error processing ${filePath}: ${(error as Error).message}`);
+  }
+};
+
+const processInChunks = async <T>(
+  items: T[],
+  processItem: (item: T) => Promise<void>,
+  chunkSize: number,
+): Promise<void> => {
+  for (let i = 0; i < items.length; i += chunkSize) {
+    const chunk = items.slice(i, i + chunkSize);
+    await Promise.all(chunk.map(processItem));
+  }
+};
+
+const watchFiles = async (directory: string): Promise<void> => {
+  const watcher = chokidar.watch(["**/*.md", "**/*.mdx"], {
+    ignored: [
+      "**.**",
+      /(^|[\/\\])\../,
+      "**/node_modules/**",
+      "**/.git/**",
+      ".gitignore",
+    ], // ignore dotfiles, node_modules, .git, and .gitignore
+    persistent: true,
+    cwd: directory,
+  });
+
+  console.log("Watch mode started. Waiting for file changes...");
+
+  watcher
+    .on("add", filePath => processFile(path.join(directory, filePath)))
+    .on("change", filePath => processFile(path.join(directory, filePath)))
+    .on("unlink", filePath => console.log(`File ${filePath} has been removed`));
+};
+
+const main = async (): Promise<void> => {
```

Comment:
with the addition of the udpated contributing doc, the coder will attempt to parse the CONTRIBUTING.md file, which results in an error and the build to fail